hal/armv7a/exceptions.c

```diff
@@ -49,18 +50,28 @@ struct {
 enum { exc_reset = 0, exc_undef, exc_svc, exc_prefetch, exc_abort };
 
 
-void hal_exceptionsDumpContext(char *buff, exc_context_t *ctx, int n)
+const char *hal_exceptionMnemonic(int n)
 {
 	static const char *const mnemonics[] = {
-		"0 #Reset",       "1 #Undef",    "2 #Syscall",    "3 #Prefetch",
-		"4 #Abort",       "5 #Reserved", "6 #FIRQ",       "7 #IRQ"
+		"0 #Reset", "1 #Undef", "2 #Syscall", "3 #Prefetch",
+		"4 #Abort", "5 #Reserved", "6 #FIRQ", "7 #IRQ"
 	};
+
+	if (n < 0 || n >= sizeof(mnemonics) / sizeof(mnemonics[0])) {
+		return "Unknown exception";
+	}
+	return mnemonics[n];
+}
+
+
+void hal_exceptionsDumpContext(char *buff, exc_context_t *ctx, int n)
+{
 	size_t i = 0;
 
 	n &= 0x7;
```

Comment:
This line is unnecessary.
You check in lines 60-62.

Reply:
I don't know if there can't be some other information encoded in higher bits, if not here then on other architectures. Will move this to line 60 and get rid of if statement.